test/unit/buffer_spec.lua

```diff
@@ -431,31 +431,16 @@ describe('buffer functions', function()
       'aaaa%=b%=c%=d%=e%=fg%=hi%=jk%=lmnop%=qrstuv%=wxyz',
       'aaaa  b  c  d  e  fg  hi  jk  lmnop  qrstuv   wxyz')
 
-    -- maximum stl item testing
-    statusline_test('should handle a much larger amount of = than buffer locations', 20,
-      ('%='):rep(STL_MAX_ITEM - 1),
-      '                    ') -- Should be fine, because within limit
-    statusline_test('should handle a much larger amount of = than stl max item', 20,
-      ('%='):rep(STL_MAX_ITEM + 1),
-      '                E541') -- Should show the VIM error
+    -- stl item testing
+    statusline_test('should handle a larger amount of = than stl initial item', 20,
+      ('%='):rep(STL_INITIAL_ITEMS * 5),
+      '                    ') -- Should not show any error
     statusline_test('should handle many extra characters', 20,
-      'a' .. ('a'):rep(STL_MAX_ITEM * 4),
-      '<aaaaaaaaaaaaaaaaaaa') -- Does not show the error because there are no items
-    statusline_test('should handle almost maximum of characters and flags', 20,
-      'a' .. ('%=a'):rep(STL_MAX_ITEM - 1),
-      'a<aaaaaaaaaaaaaaaaaa') -- Should not show the VIM error
+      'a' .. ('a'):rep(STL_INITIAL_ITEMS * 5),
+      '<aaaaaaaaaaaaaaaaaaa') -- Does not show any error
     statusline_test('should handle many extra characters and flags', 20,
-      'a' .. ('%=a'):rep(STL_MAX_ITEM),
-      'a<aaaaaaaaaaaaa E541') -- Should show the VIM error
-    statusline_test('should handle many extra characters and flags', 20,
-      'a' .. ('%=a'):rep(STL_MAX_ITEM * 2),
-      'a<aaaaaaaaaaaaa E541') -- Should show the VIM error
-    statusline_test('should handle many extra characters and flags with truncation', 20,
-      'aaa%<' .. ('%=a'):rep(STL_MAX_ITEM),
-      'aaa<aaaaaaaaaaa E541') -- Should show the VIM error
-    statusline_test('should handle many characters and flags before and after truncation', 20,
-      'a%=a%=a%<' .. ('%=a'):rep(STL_MAX_ITEM),
-      'aaa<aaaaaaaaaaa E541') -- Should show the VIM error
```

Comment:
When does the error appear?

Reply:
The error can't appear anymore. It has vanished. Items will grow as needed so there is no limit other than available memory.

Reply:
Does (neo)vim crash if you use 300 items?

https://github.com/neovim/neovim/pull/13130#issue-507925133

Reply:
The tests had already been updated to test more than the initial capacity (100), but I've updated them to go to a 1000 highlight + items.